packages/expo-navigation-bar/src/NavigationBar.ts

```diff
@@ -1,133 +1,39 @@
-import { Platform, type EventSubscription, UnavailabilityError } from 'expo-modules-core';
-import { useEffect, useState } from 'react';
-import { processColor } from 'react-native';
+import { Platform } from 'react-native';
 import { isEdgeToEdge } from 'react-native-is-edge-to-edge';
 
-import ExpoNavigationBar from './ExpoNavigationBar';
-import {
-  NavigationBarButtonStyle,
-  NavigationBarBehavior,
-  NavigationBarPosition,
-  NavigationBarVisibility,
-  NavigationBarVisibilityEvent,
-} from './NavigationBar.types';
+import { setVisibilityAsync as originalSetVisibilityAsync } from './NativeNavigationBarWrapper';
+import { NavigationBarStyle, NavigationBarVisibility } from './NavigationBar.types';
 
-let shouldWarnAboutEdgeToEdge = Platform.OS === 'android' && isEdgeToEdge();
+export * from './NavigationBar.types';
 
-function potentiallyWarnAboutEdgeToEdge() {
-  if (shouldWarnAboutEdgeToEdge) {
-    shouldWarnAboutEdgeToEdge = false; // warn once
-    console.warn(
-      'Using expo-navigation-bar in apps with edge-to-edge layout enabled may cause unexpected behavior. Instead, use the SystemBars component from react-native-edge-to-edge. Learn more: https://expo.fyi/edge-to-edge-system-bars'
-    );
-  }
-}
+// This line only imports the type information for TypeScript type checking.  It
+// doesn't import the actual module in the compiled JavaScript code.  The actual
+// module is imported conditionally with require() below, in order to avoid
+// importing the module if edge-to-edge is not enabled (which could throw if
+// it's not linked).
+let SystemBars: typeof import('react-native-edge-to-edge').SystemBars | null = null;
 
-/**
- * Observe changes to the system navigation bar.
- * Due to platform constraints, this callback will also be triggered when the status bar visibility changes.
- *
- * @example
- * ```ts
- * NavigationBar.addVisibilityListener(({ visibility }) => {
- *   // ...
- * });
- * ```
- */
-export function addVisibilityListener(
-  listener: (event: NavigationBarVisibilityEvent) => void
-): EventSubscription {
-  if (__DEV__) {
-    potentiallyWarnAboutEdgeToEdge();
-  }
-  // Assert so the type is non-nullable.
-  if (!ExpoNavigationBar.addListener) {
-    throw new UnavailabilityError('NavigationBar', 'addVisibilityListener');
-  }
-  return ExpoNavigationBar.addListener('ExpoNavigationBar.didChange', listener);
+if (isEdgeToEdge() && Platform.OS === 'android') {
```

Comment:
```suggestion
const isEdgeToEdgeEnabled = Platform.OS === 'android' && isEdgeToEdge();

if (isEdgeToEdgeEnabled) {
```
We can consider exporting `isEdgeToEdgeEnabled` as our public API too

Reply:
I'm not a fan of this change, edge-to-edge is always true on iOS, but this const will be false for iOS, which may be misleading for maintainers in the future

Reply:
In the end I did something similar, but called it `canRunEdgeToEdgeMethods`. I didn't export it though